File: components-core/src/main/java/com/adyen/checkout/components/GenericComponentState.java
/*
 * Copyright (c) 2021 Adyen N.V.
 *
 * This file is open source and available under the MIT license. See the LICENSE file for more info.
 *
 * Created by josephj on 21/1/2021.
 */

package com.adyen.checkout.components;

import androidx.annotation.NonNull;

import com.adyen.checkout.components.model.payments.request.PaymentComponentData;
import com.adyen.checkout.components.model.payments.request.PaymentMethodDetails;

public class GenericComponentState<PaymentMethodDetailsT extends PaymentMethodDetails> extends PaymentComponentState<PaymentMethodDetailsT> {

    /**
     * PaymentComponentState for any component without additional data.
     */
    public GenericComponentState(
            @NonNull PaymentComponentData<PaymentMethodDetailsT> paymentComponentData,
            boolean isInputValid,
            boolean isReady
    ) {
        super(paymentComponentData, isInputValid, isReady);
    }
}

File: core/src/main/java/com/adyen/checkout/core/components/internal/ui/state/GenericComponentState.kt
/*
 * Copyright (c) 2026 Adyen N.V.
 *
 * This file is open source and available under the MIT license. See the LICENSE file for more info.
 *
 * Created by josephj on 11/8/2026.
 */

package com.adyen.checkout.core.components.internal.ui.state

import androidx.annotation.RestrictTo

@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
data class GenericComponentState(
    val isLoading: Boolean,
) : ComponentState
